lib/order.rb

```diff
@@ -7,12 +35,78 @@ def initialize(id, products)
       @products = products
     end
 
+    def self.all
+      orders_array = []
+      ind_order_array = []
+
+      # csv = CSV.read("support/orders.csv")
+      # ap csv.length
+      CSV.open("support/orders.csv", 'r').each do |line|
+        # ap line
+        ind_order_array << line
+        # ap "ind_orders_array : #{ind_orders_array}"
+        id = line[0].to_i
+        product_array = line[1].split(';') # array of product:price
+
+        items = {}
+        product_array.each do |element_pair|
+           item = element_pair.split(':') # array [0]item, [1]price
```

Comment:
I like the comments here to indicate the format you expect for these variables.

Alternatively, you could write this inner loop as follows:

```ruby
product_array.each do |element_pair|
  name, price = element_pair.split(':')
  items[name] = price.to_f
end
```